tests/backend/test_events.py

```diff
@@ -146,4 +146,29 @@ async def test_events_listen_wait_cancelled(backend_asgi_app, alice_ws):
         listen.rep_done = True
 
 
+@pytest.mark.trio
+async def test_events_close_connection_on_backpressure(
+    backend_asgi_app, alice, bob, monkeypatch, backend_authenticated_ws_factory
+):
+    # The channel has a queue of size 1, meaning it will be filled after a single command
+    monkeypatch.setattr("parsec.backend.client_context.AUTHENTICATED_CLIENT_CHANNEL_SIZE", 1)
+    # We need to manually initialize the websockets using the factory here to make
+    # sure the monkeypatch happens before the memory channel creation and thus
+    # give then a size of 1
+    async with backend_authenticated_ws_factory(backend_asgi_app, alice) as alice_ws:
+        async with backend_authenticated_ws_factory(backend_asgi_app, bob) as bob_ws:
```

Comment:
Note you can directly use `bob_ws` fixture instead of creating it here. This is because `bob_ws` is used in the test only to send commands (and hence will never use it client context's channel)